### PR TITLE
[echo] implement a TCP echo server

### PR DESCRIPTION
**Pull request by danking**
I will test this with a forthcoming PR which adds a TCP gateway.

### CODE REVIEW

File: echo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM gcr.io/{{ global.project }}/ubuntu:18.04
```
Comment (jigold) [inline echo/Dockerfile:1]: Depending on whether #9565 goes in first, we'll need to fix this.

**Review by jigold — APPROVED** [review]

**Comment by jigold** [comment]
Does this need to have the service with the router like all the other services?

**Review by jigold — CHANGES_REQUESTED** [review]
see comment

**Comment by danking** [comment]
No, it's not an HTTP service, it's plain old TCP, so `router` (which is an HTTP router) won't work here. The TCP gateway / router will come next.

**Review by jigold — APPROVED** [review]

**Pull request merged**